source/js/interview.js

```diff
@@ -1,277 +1,258 @@
-                if(hashCode(card.id) == id)
-                    loadcard = card;
-
-                if(hashCode(card.id) == hashCode(id))
-                    loadcard = card;
-
-                
-                if(hashCode(card.id.toLowerCase()) == hashCode(id.toLowerCase()))
-                    loadcard = card;
-            }
-
-            if(loadcard === null)
-                console.error(`Provided ID [${id}] seems to be invalid!`);
-            else {
-                if(!isCardOpen)
-                    document.getElementById(loadcard.id).click();
-                else 
-                    changeCard(loadcard);
-            }
-        }
-        return;
-    }
+    return h;
+  }
+
+  // Function that's responsible for checking the URL for any arguments/parameters and taking actions accordingly
+  function checkUrlForPotentialParameters() {
+    let loadcard = null;
+    // Getting current URL
+    const url = new URL(window.location.href);
+    if (!url.search || url.search == null) { return; }
+    // Trying to get parameter id from our URL, which could look like this (https://www.galileoschool.sk/interviews/index.html?id=518421)
+    // evertyhing behind the "?" symbol is considered a parameter
+    // Multiple parameters are divided with "&" symbol ex. (https:/www.galileoschool.sk/interviews/index.html?id=518421&showPhotos=false)
+    // I decided to use id because each interview card has its own UID and it's easier to compare things if you know what you are comparing
+    const id = url.searchParams.get('id');
+    if (id && id != null) {
+      for (const card of cards) {
+        if (card.id == 'modal-box') { break; }
+
+        if (hashCode(card.id) == id) { loadcard = card; }
 
-    // Function that's responsible for modifying the URL without reloading the page
-    function modifyUrl(id) {
-        if(window.location.href.indexOf("id=") !== -1)
-            var build_url = window.location.href.replace(window.location.href.split("id=")[1], id);
-        else
-            var build_url = (window.location.href + "?id=" + id);
-        
-        if(id == null)
-            build_url = build_url.split("?")[0];
-
-        // This simply changes the current URL without reloading the page, by creating and activating another history entry without firing the haschanged event for the browser.
-        history.pushState({
-            id: 'interview'
-        }, 'GALILEO | INTERVIEWS', build_url);
+        if (hashCode(card.id) == hashCode(id)) { loadcard = card; }
+
+
+        if (hashCode(card.id.toLowerCase()) == hashCode(id.toLowerCase())) { loadcard = card; }
+      }
+
+      if (loadcard === null) { console.error(`Provided ID [${id}] seems to be invalid!`); } else if (!isCardOpen) { document.getElementById(loadcard.id).click(); } else { changeCard(loadcard); }
     }
-});
\ No newline at end of file
+  }
+
+  // Function that's responsible for modifying the URL without reloading the page
+  function modifyUrl(id) {
```

Comment:
what does the input mean? maybe also call the input variable better - not just `id` (what kind of id?)